src/lib.rs

```diff
@@ -159,6 +159,14 @@ impl MiniscriptKey for bitcoin::PublicKey {
     }
 }
 
+impl MiniscriptKey for bitcoin::schnorr::PublicKey {
+    type Hash = hash160::Hash;
+
+    fn to_pubkeyhash(&self) -> Self::Hash {
```

Comment:
The question is why we need to support pubkey hashes in Taproot - and why it is of hash160::Hash type, when we have specific hash types for other cases (like `PubkeyHash`, `WPubkeyHash`)?

Reply:
Good call. Basically, when this was originally implemented I was not aware of specific hash types in rust-bitcoin. I will change bitcoin::schnorr::PublicKey in this PR and change the `impl MiniscriptKey for bitcoin::PublicKey` in a separate PR.

Reply:
Thanks!But in whic cases do we need pubkey hashes in Taproot/script? I understand that nothing prohibits from adding key hash to arbitrary tapscript leaf, but I think we should discourage this nonstandard practice in API and just handle that as any other hash, unrelated to pubkey

Reply:
This is more a spec question. I don't see why we should disallow `pkh`. It's more efficient in case the fragment is not satisfied.

Reply:
As in `DUP HASH160 <hash> EQUALVERIFY CHECKSIG` is still a valid script and has less size than `<x_only_pk> CHECKSIG`. So, there is one use-case

Reply:
But would this smaller size matter for the branches of the tapscript which are never used? It will never hit blockchain. But while spent, it will have a larger size for `witness` in the spending tx input.

Reply:
Consider the following:
`thresh(11,S1,S2,S2......,S15)` where `S15` is `or(pk(likely),pkh(unlikely))`

It is difficult to always decompose scripts into ORs(DNF). Therefore, it might be necessary to a leaf miniscript that contains a or